### PR TITLE
Hotfix CSS bugs in Safari, IE11, and Firefox for IBM Cloud

### LINKED ISSUE #9: Typeahead clear icon misaligned in Firefox, Safari, and IE11
Firefox:
clear-icon_firefox

Safari:
clear-icon_safari

IE11:
clear-icon_ie11

### LINKED ISSUE #10: Typeahead component does not close correctly in IE11 after making a selection
Steps to reproduce:
1) Type in the component
2) Click on an item

Result:
Items in list go away, but a large white box remains where the items used to be.

Expected behavior:
The white box collapses when the typeahead closes and the items go away.

Note: Clicking on the close button multiple times will eventually make the white box disappear.

screen shot 2017-10-26 at 11 38 30 am

### LINKED ISSUE #11: Multiple clear icons displayed in IE11
Steps to reproduce:
1) Click on the Typeahead to give it focus
2) Type some text

Result:
The Typeahead input displays an "X" icon to the left of the close icon.

Expected behavior:
After completing the steps above, only the close icon is displayed

Note:
If you click anywhere to remove focus from the Typeahead component, the "X" goes away

multiple-clear-icons

### PR DESCRIPTION
**Pull request by joshblack**
Closes #9, Closes #10, Closes #11 

#### Changed

- Add pseudo element styles for `-ms-clear` icon that shows up to clear suggestions
- Update focus styles to include active sometimes for Firefox
- Update how items are centered visually in the container
- Fix IE11 issue where item list would have a height of `-0.01` which caused it to expand infinitely.

#### Removed

### CODE REVIEW

File: components/Typeahead/_typeahead.scss [hunk 1]
```diff
@@ -6,72 +6,108 @@
...
-    right: 45px;
-    cursor: pointer;
+  .bx--typeahead__input {
+    padding-right: rem(70px);
+  }
+
+  // Apparently, in IE10+ there is a "clear" icon added to inputs that we
+  // need manually hide since we use our own icon to clear.
+  .bx--typeahead__input::-ms-clear {
+    display: none;
```
Comment (mven) [inline components/Typeahead/_typeahead.scss:18]: Should this also have a `visibility: hidden` for a11y?

**Comment by marijohannessen** [comment]
Looks good to me!

File: components/Typeahead/_typeahead.scss (hunk 1 above)
Comment (joshblack) [reply components/Typeahead/_typeahead.scss:18]: Sure! Will add in.

Comment (tw15egan) [reply components/Typeahead/_typeahead.scss:18]: It seems like they both accomplish the same thing in regards to screen readers, but might as well add it just to be safe. 

https://webaim.org/techniques/css/invisiblecontent/

**Review by tw15egan — APPROVED** [review]

**Review by mven — APPROVED** [review]

**Pull request merged**